python/instrumentation/openinference-instrumentation-groq/src/openinference/instrumentation/groq/_wrappers.py

```diff
@@ -0,0 +1,185 @@
+from abc import ABC
+from enum import Enum
+from typing import Any, Callable, Iterator, List, Mapping, Tuple
+
+import opentelemetry.context as context_api
+from openinference.instrumentation import get_attributes_from_context, safe_json_dumps
+from openinference.semconv.trace import (
+    EmbeddingAttributes,
+    OpenInferenceMimeTypeValues,
+    OpenInferenceSpanKindValues,
+    SpanAttributes,
+)
+from opentelemetry import trace as trace_api
+from opentelemetry.util.types import AttributeValue
+
+
+def _flatten(mapping: Mapping[str, Any]) -> Iterator[Tuple[str, AttributeValue]]:
+    for key, value in mapping.items():
+        if value is None:
+            continue
+        if isinstance(value, Mapping):
+            for sub_key, sub_value in _flatten(value):
+                yield f"{key}.{sub_key}", sub_value
+        elif isinstance(value, List) and any(isinstance(item, Mapping) for item in value):
+            for index, sub_mapping in enumerate(value):
+                for sub_key, sub_value in _flatten(sub_mapping):
+                    yield f"{key}.{index}.{sub_key}", sub_value
+        else:
+            if isinstance(value, Enum):
+                value = value.value
+            yield key, value
+
+
+class _WithTracer(ABC):
+    """
+    Base class for wrappers that need a tracer.
+    """
+
+    def __init__(self, tracer: trace_api.Tracer, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._tracer = tracer
+
+
+class _CompletionsWrapper(_WithTracer):
+    """
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+
+    def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        llm_invocation_params = kwargs
+        llm_messages = dict(kwargs).pop("messages", None)
+        if context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY):
+            return wrapped(*args, **kwargs)
+
+        # Prepare invocation parameters by merging args and kwargs
+        invocation_parameters = {}
+        for arg in args:
+            if arg and isinstance(arg, dict):
+                invocation_parameters.update(arg)
+        invocation_parameters.update(kwargs)
+
+        span_name = "Completions"
+        with self._tracer.start_as_current_span(span_name, attributes={}) as span:
```

Comment:
i learned this from my last PR but since we're setting exceptions manually below, we should set the `record_exception` and `set_status_on_exception` to false

example:
https://github.com/Arize-ai/openinference/blob/main/python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py#L150-L151